task5.js

```diff
@@ -0,0 +1,74 @@
+/*  Завдання - 5:
+ 
+Виведіть на екран властивості first та last з вкладеного об'єкта name. 
+Змініть значення властивості balance на '$2000'.
+Видаліть властивість email.
+Виведіть на екран весь зміст об'єкта за допомогою кода console.log(user) */
+
+let user = {
+    balance: '$1,250.89',
+    age: 24,
+    name: {
+        first: 'Golden',
+        last: 'Clements'
+    },
+    company: 'EWAVES',
+    email: 'golden.clements@ewaves.io',
+    friends: [
+        {
+            id: 0,
+            name: 'Buchanan Austin'
+        },
+        {
+            id: 1,
+            name: 'Richmond Garrison'
+        },
+        {
+            id: 2,
+            name: 'Burns Cook'
+        },
+        {
+            id: 3,
+            name: 'Sally Mcpherson'
+        }
+    ],
+    favoriteFruit: 'strawberry',
+
+    // Method to return full name
+    fullName: function () { 
+        return `User name is: ${this.name.first} ${this.name.last}`;
+    },
+    //Method to update balance
+    setBalance: function (newBalance) {  
+        return this.balance = newBalance 
+    },
+    // Method to delete key in object 
+    deleteByKey: function (key) {   
+        if (key in this) {
+            if (typeof this[key] === 'function') {   
+                console.log(`Key "${key}" is a function of Object, and can't be deleted`);
+            } else {
+                delete this[key];    // Delete key in Object
+                console.log(`Key "${key}" was successfully deleted`);  
+            }
+        } else {
+            console.log('Key is missing');  
```

Comment:
Usage of error functions for object method definition can be more clear and readable

example:

> fullName: () => `User name is: ${this.name.first} ${this.name.last}`,
setBalance: (newBalance) => this.balance = newBalance,
deleteByKey: (key) => {
    if (key in this) {
        if (typeof this[key] === 'function') {
            console.log(`Key "${key}" is a function of Object, and can't be deleted`);
        } else {
            delete this[key];
            console.log(`Key "${key}" was successfully deleted`);
        }
    } else {
        console.log('Key is missing');
    }
}

Reply:
Arrow functions don’t have their own this, so they can’t easily access the object they belong to.

 If we want to use arrow functions and still access the object’s properties, we need to use the .bind() method to set this correctly. 

But usually, it’s easier to use regular functions in objects, because they automatically have the correct this context.